packages/atomic/src/components/insight/user-actions/atomic-insight-user-action-session/atomic-insight-user-actions-session.tsx

```diff
@@ -0,0 +1,165 @@
+import {Component, h, Prop, Watch, State} from '@stencil/core';
+import Bookmark from '../../../../images/bookmark.svg';
+import ThreeDotsIcon from '../../../../images/three-dots.svg';
+import {parseTimestampToDateDetails} from '../../../../utils/date-utils';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+
+export type UserActionType =
+  | 'SEARCH'
+  | 'CLICK'
+  | 'TICKET_CREATION'
+  | 'VIEW'
+  | 'CUSTOM';
+
+type UserActions = Array<{
+  type: UserActionType;
+  origin: string;
+  timestamp: number;
+  actionTitle: string;
+}>;
+
+/**
+ * @internal
+ * The `AtomicInsightUserActionSession` component displays all the user actions that took place during a specific user session.
+ * @category Insight Panel
+ * @example
+ * <atomic-insight-user-actions-session userActions={actions} startDate="1723035731"></atomic-insight-user-actions-session>
+ */
+@Component({
+  tag: 'atomic-insight-user-actions-session',
+  styleUrl: 'atomic-insight-user-actions-session.pcss',
+  shadow: true,
+})
+export class AtomicInsightUserActionsSession {
+  @InitializeBindings() public bindings!: InsightBindings;
+  @State() public error!: Error;
+
+  @Prop({mutable: true}) public startDate!: number;
+  @Prop({mutable: true}) public userActions: UserActions = [];
+
+  private caseCreationIndex = -1;
+  private userActionsToDisplay: UserActions = [];
+  private userActionsAfterCaseCreation: UserActions = [];
+  private showMoreActionsButtonShouldBeVisible = false;
+  @State() actionsAfterCaseCreationShouldBeVisible = false;
+
+  connectedCallback() {
+    this.setUserActionsToDisplay();
+    this.extractUserActionsAfterCaseCreation();
+  }
+
+  setCaseCreationIndex() {
+    this.caseCreationIndex = this.userActions.findIndex(
+      (action) => action.type === 'TICKET_CREATION'
+    );
+  }
+
+  @Watch('userActions')
+  setUserActionsToDisplay() {
+    this.setCaseCreationIndex();
+
+    if (this.caseCreationIndex === -1) {
```

Comment:
Now that ticket creation is mandatory, I don't think there should be a situation where the index is not found. 
In the preprocessing PR, if there is no ticketCreationDate given, it returns an empty timeline anyways so there won't be userActions to display. 

Otherwise if one is given, we will create the current session no matter what.

Reply:
We need this regardless.

Cause this component displays any kind of sessions, it can be a following session, a preceding session or an active session.
so basically this check is now useful to know if the session is an active session that contains a caseCreation or a following/preceding session that does not contain the case creation.